financial/exponential_moving_average.py

```diff
@@ -0,0 +1,72 @@
+"""
+    Calculates exponential moving average (EMA) on the series of numbers
+    Wikipedia Reference: https://en.wikipedia.org/wiki/Exponential_smoothing
+    Reference: https://www.investopedia.com/terms/e/ema.asp#toc-what-is-an-exponential-moving-average-ema
+
+    Exponential moving average is used in finance to analyze changes stock prices.
+    EMA is used in conjunction with Simple moving average (SMA), EMA reacts to the
+    changes in the value quicker than SMA, which is one of the advantages of using EMA.
+"""
+
+from collections.abc import Iterator
+
+
+def exponential_moving_average(
+    series_generator: Iterator[float], window_size: int
+) -> Iterator[float]:
+    """
+    Returns the generator which generates exponential moving average of the given
+    series generator
+    >>> list(exponential_moving_average((ele for ele in [2, 5, 3, 8.2, 6, 9, 10]), 3))
```

Comment:
Just easier to read...
```suggestion
    >>> list(exponential_moving_average(iter([2, 5, 3, 8.2, 6, 9, 10]), 3))
```